test/unit/error.test.js

```diff
@@ -47,6 +49,46 @@ describe('MongoErrors', () => {
     });
   }
 
+  describe('MongoError#constructor', () => {
```

Comment:
Would these assertions be better added to the loop over all the errors here: https://github.com/mongodb/node-mongodb-native/blob/main/test/unit/error.test.js#L34-L48

Reply:
I thought about this, but I feel like it's kind of awkward in there, because we can't really recursively define error constructor behavior outside the one shared name property, so then if we do it in the loop, we have to check for the error class name, which feels super hacky.

Reply:
Fair enough, let's keep this then!